### ISSUE TITLE
Mirroring Magic Trackpad preference “On” does not stay active.

### ISSUE THREAD

**Issue by jasonology** [folivoraAI/BetterTouchTool#935]
**IMPORTANT:** Don't post private requests here (e.g. license issues, etc.). Please send me an email at boastr.net@gmail.com instead.

---

#### Description of bug/feature request/question:

The option ‘Mirror complete Magic Trackpad orientation for “upside-down” usage’ keeps turning itself off.

#### Affected input device (e.g. MacBook Trackpad, Magic Mouse/Trackpad, Touch Bar, etc.):

Magic Trackpad (original)

#### Device information:

Type of Mac: MacBook Pro (Retina, 13-inch, Early 2015)
macOS version: 10.12.4
BetterTouchTool version: 2.11

#### Additional information (e.g. StackTraces, related issues, screenshots, workarounds, etc.):

This seems to happen when the Trackpad goes to sleep or is restarted, but happens other times too.
The tickbox remains ticked in the GUI, I have to untick it and retick it for the function to work.

---

**Comment by abertheham** [comment]
Same issue here.  Most notable correlations/causes:

-  essentially every time the computer and/or trackpad go to sleep
- while opening BTT preferences (from the menubar icon as well as using a gesture)
- 4-finger gestures; in particular one of my custom tap sequences (see below) disables mirroring almost every single time I use it
![screen shot 2017-06-03 at 14 53 36](https://cloud.githubusercontent.com/assets/18314254/26756677/d790bd0e-486d-11e7-8eb4-8b2d7e474cb1.png)


And as stated above, when I look at BTT, the box is still ticked.  So I can either untick/retick the box, or - my most recent work around - set a quick and easy gesture to restart BTT.

Devices:
Macbook Pro (Non-Retina, 13-inch, Mid 2012)
macOS Sierra 10.12.5
BTT 2.20
MagicTrackpad (original)


Thanks for any help with this issue.  There's almost no other way for people to use their trackpads upside down any more - you're the only hope for ergonomic decency.

**Comment by mauigirl** [comment]
Same loss of trackpad mirroring. I have disabled the menu bar icon but still happening. The mirroring function fales 100% upon sleep/wake but also randomly while working through the day. I have to go back into advanced trackpad settings, untick and retick the mirroring box numerous times per day.

Mac OS 10.12.5
MBP Mid-2012
2.7 Intel Core i7
16GB RAM

**Comment by mauigirl** [comment]
@abertheham could you share your restart gesture?

**Comment by abertheham** [comment]
@mauigirl Sure... I use a 4 finger double tap, but anything that's easy to trigger regardless of whether-or-not the trackpad is "upside-down" (i.e. symmetric taps/clicks) should do the trick.  

Supplemental note: In addition to the observations I made above, I've also noticed that any gesture which includes one or more "delay next action" is also more likely to cause the problem.

Just in case it's helpful to @fifafu or anyone else, here's all of my current default/global gesture configs.

![image](https://user-images.githubusercontent.com/18314254/27264421-940a9960-5443-11e7-8a35-6d4d493725e4.png)

**Comment by mauigirl** [comment]
Thank you @abertheham ! That's perfect. FYI I have not configured any gestures until now. The sole purpose of BTT for me was the trackpad mirroring, so all other options are set to default (except for removing menu bar icon and disabling window snapping).

**Comment by derekvan** [comment]
I'm having this same problem. It's not clear what causes the mirroring to quit working: no patterns I can discern.

**Comment by fifafu** [comment]
Sorry for not being able to look into this earlier.
Does this still happen? 

Does the checkbox really become unchecked in BTT?

**Comment by mauigirl** [comment]
Yes it still happens every day. It occurs 100% upon wake from sleep and
then several times randomly throughout the day even while actively working.
I don't know if the box becomes unchecked as I am just using a 4-finger
command to relaunch BTT (which works 90%) of the time. I think the other
10% BTT itself is crashing because the command doesn't always work and I
have to launch BTT from the Finder. I will try and remember to check
Activity Monitor in the future to confirm if it's still running or not.

Thank you!

*Jennifer Androes*


On Sun, Nov 19, 2017 at 9:20 AM, Andreas Hegenberg <notifications@github.com

**Comment by mauigirl** [comment]
I looked this morning (trackpad wasn't mirroring properly upon wake but BTT
services were running) and the checkbox was still checked even though it
wasn't mirroring. I unchecked and checked again and then it started working
properly. Which is similar I guess to us restarting BTT with a gesture and
having it work. Hope that helps.

*Jennifer Androes*

**Comment by fifafu** [comment]
I have been using the mirrored trackpad for a few months now and couldn't find an issue.
It may become unmirrored for a few seconds when BTT restarts itself (which usually happens after sleep), but for me it always mirrored correctly again afterwards.

If this is still happening to you please comment here, for now I'm closing this.

**Comment by mauigirl** [comment]
It is STILL happening daily during regular use, not just wake/sleep. I
would expect the wake/sleep behaviour but two additional things frequently
happen:

1) BTT crashes (the icon disappears from the menu bar) and trackpad reverts
orientation. i relaunch btt using Spotlight as it's the easiest way to
navigate without a working trackpad and then all is well

2) BTT is still running (the icon remains on the menu bar) but trackpad
reverts orientation. here i do a programmed 4-finger restart of btt and all
is well

This happens 2-3 times per day, in various apps without warning or apparent
pattern. And sometimes when switching between apps. I would not consider
this issue closed at all. I perform BTT updates whenever they roll through
and still the problem persists.

Thank you!


*Jennifer Androes*


On Fri, Feb 2, 2018 at 11:10 PM, Andreas Hegenberg <notifications@github.com

**Comment by mauigirl** [comment]
It is STILL happening daily during regular use, not just wake/sleep. I
would expect the wake/sleep behaviour but two additional things frequently
happen:

1) BTT crashes (the icon disappears from the menu bar) and trackpad reverts
orientation. i relaunch btt using Spotlight as it's the easiest way to
navigate without a working trackpad and then all is well

2) BTT is still running (the icon remains on the menu bar) but trackpad
reverts orientation. here i do a programmed 4-finger restart of btt and all
is well

This happens 2-3 times per day, in various apps without warning or apparent
pattern. And sometimes when switching between apps. I would not consider
this issue closed at all. I perform BTT updates whenever they roll through
and still the problem persists.

Thank you!

*Jennifer Androes*


On Fri, Feb 2, 2018 at 11:10 PM, Andreas Hegenberg <notifications@github.com

**Comment by derekvan** [comment]
Also still happening for me daily. What seems like I notice is that the computer will hang for a second, the beach ball will spin, and the pointer will now be un-mirrored. This happens in lots of different apps, it's not just one or two interfering with BTT. I most typically see condition number 2 listed by Jennifer above.

**Comment by jasonology** [comment]
Still happening on 2.318. Updating now, but this bug is not listed in the release notes.

**Comment by fifafu** [comment]
I'll reopen here but I don't think I can do anything here currently. It seems to be working correctly on all my systems  :-(

As this is using undocumented api it's possible that this doesn't behave correctly on all systems unfortunately.

**Comment by mparmer** [comment]
Hello, I'm a new user of BTT, using it for the mirrored trackpad functionality with a MBP2017.. I am also experiencing a problem with the mirroring functionality.  If i press the key combination cmd + opt, it disables the mirroring and I have to go into BTT settings and unclick/reclick the mirroring box to get it working again.  This is a big problem, as there are several other programs already using this key combo, ie chrome developer tools is accessed through cmd + opt + i.  Please is there a way to disable this key combo from borking the config?

**Comment by fifafu** [comment]
that's weird, it doesn't do that here. Do you maybe have BetterTouchTool set to disable when cmd+opt is pressed? (In the advanced preferences)

**Comment by mparmer** [comment]
I just figured out that it's conflicting with an app called "Flexiglass" that is a window manager for OSX.  I had disabled all the window management stuff in BTT, thinking that it would be fine to use them both, but as it turns out they conflict with each other.  I have replicated the key combos I use in BTT, and it's working fine now!  Thanks for the quick response, it allowed me to move on to troubleshooting other things and ultimately finding the solution!
